src/Kdyby/Translation/PrefixedTranslator.php

```diff
@@ -49,6 +49,10 @@ public function __construct($prefix, ITranslator $translator)
 
 	public function translate($message, $count = NULL, $parameters = array(), $domain = NULL, $locale = NULL)
 	{
+		if ($message instanceof Phrase) {
+			return $message->translate($this);
```

Comment:
That wouldn't have been a problem either :) You would just create some `->copyPrefixed($prefix)` method that would do this inside the `Phrase` object. But since it's a value object, I would preffer having it less logic :)